algorithms.py

```diff
@@ -23,11 +24,11 @@ def hill_climb(problem, max_iters=np.inf, restarts=0, init_state=None):
     optimizes fitness function.
     best_fitness: float. Value of fitness function at best state
     """
-    if (type(max_iters) != int and max_iters != np.inf \
-       and max_iters.is_integer() == False) or (max_iters < 0):
+    if (not isinstance(max_iters, int) and max_iters != np.inf
```

Comment:
Good suggestion. I was previously unaware of the isinstance() function.

Reply:
Thank you.